Source/WebKit/Platform/IPC/Connection.cpp

```diff
@@ -171,7 +171,7 @@ Ref<Connection::SyncMessageState> Connection::SyncMessageStateImpl::forCurrentTh
 
 void Connection::SyncMessageStateImpl::enqueueMatchingMessages(Connection& connection, MessageReceiveQueue& receiveQueue, const ReceiverMatcher& receiverMatcher)
```

Comment:
So these changes are testable in TestIPC, so you can create bunch of message sending tests that start new runloops and send messages back and forth.
The tests could test new runloops in different threads and in same threads.

Reply:
Another test set needs to be such that you write a mock serialfunctiondispatcher that detects whether or not it is accessed incorrectly.
E.g. one thread sends incorrect messages
the receiver marks the connection invalid at time point a.
the receiver uses the mock function dispatcher to detect that the function dispatcher was not accessed after invalidate().

E.g. before this patch it didn't make a difference, because the thing only used main runloop which is always around. Now the function dispatcher might go away, and as such the access semantics must be precise (which they are not until the bugs are fixed)..